projects/backend/utils/article.ts

```diff
@@ -15,4 +17,99 @@ const articleShouldHaveDropCap = ({
 const isHTMLElement = (el?: BlockElement): el is HTMLElement =>
     !!el && el.id === 'html'
 
-export { articleShouldHaveDropCap, isHTMLElement }
+type CAPIEndpoint = 'preview' | 'printsent' | 'live'
+
+const getEndpoint = (capi: CAPIEndpoint, paths: string[]): string => {
+    const queryString = `?ids=${paths.join(',')}&api-key=${
+        process.env.CAPI_KEY
+    }&format=thrift&show-elements=all&show-atoms=all&show-rights=all&show-fields=all&show-tags=all&show-blocks=all&show-references=all&page-size=100`
+
+    switch (capi) {
+        case 'printsent':
+            return `${process.env.psurl}/search${queryString}`
+        case 'live':
+            return `https://content.guardianapis.com/search${queryString}`
+        case 'preview':
+            return `${process.env.capiPreviewUrl}/search${queryString}`
+        default:
+            return ''
+    }
+}
+
+const getStsCreds = async (role: string) => {
+    const sts = new STS({ apiVersion: '2011-06-15' })
+    const creds = await sts
+        .assumeRole({
+            RoleArn: role as string,
+            RoleSessionName: 'capi-assume-role-access2',
+        })
+        .promise()
+        .catch(err => console.error('assume role failed', err))
+
+    if (creds && creds.Credentials) {
+        return {
+            secretAccessKey: creds.Credentials.SecretAccessKey,
+            accessKeyId: creds.Credentials.AccessKeyId,
+            sessionToken: creds.Credentials.SessionToken,
+        }
+    } else {
+        const errorMessage = `Could not generate credentials using STS role ${role}`
+        console.error(errorMessage)
+        throw new Error(errorMessage)
+    }
+}
+
+const getProfileCreds = async () => {
+    const credentials = new SharedIniFileCredentials({ profile: 'capi' })
+    return {
+        secretAccessKey: credentials.secretAccessKey,
+        accessKeyId: credentials.accessKeyId,
+        sessionToken: credentials.sessionToken,
+    }
+}
+
+/**
+ * To access the capi Preview endpoint we need to sign requests (as it is a private api gateway endpoint)
+ * This function generates the necessary headers.
+ * @param endpoint
+ */
+async function sign(endpoint: string) {
+    const url = new URL(endpoint)
+
+    const opts = {
+        region: 'eu-west-1',
+        service: 'execute-api',
+        host: url.hostname,
+        path: url.pathname + url.search,
+    }
+
+    const credentials = process.env.capiAccessArn
+        ? await getStsCreds(process.env.capiAccessArn)
+        : await getProfileCreds()
+
+    const { headers } = new RequestSigner(opts, credentials).sign()
+
+    // console.log(`Signed request, generated headers: ${JSON.stringify(headers)}`)
```

Comment:
I have commented out this line which prints secure header, do you think it is okay to log this info @philmcmahon ?

@JamieB-gu asked why do log this seems to print capi token

Reply:
This sounds like a bad idea to print this, my fault probably left over from debugging capi url signing

Reply:
Should we just delete this line?